antenna/crashmover.py

```diff
@@ -110,18 +92,24 @@ def __init__(self, config):
             config.with_namespace("crashpublish")
         )
 
-        # Gevent pool for crashmover workers
-        self.crashmover_pool = Pool(size=self.config("concurrent_crashmovers"))
-
-        # Queue for crashmover work
-        self.crashmover_queue = deque()
-
-        # Register hb functions with heartbeat manager
-        register_for_heartbeat(self.hb_report_health_stats)
-        register_for_heartbeat(self.hb_run_crashmover)
-
-        # Register life function with heartbeat manager
-        register_for_life(self.has_work_to_do)
+        # configure retries on save and publish
+        self.crashmover_save = retry(
+            module_logger=LOGGER,
+            wait_time_generator=_incr_wait_generator(
+                counter="save_crash_exception.count",
+                attempts=self.config("max_attempts"),
+                sleep_seconds=self.config("retry_sleep_seconds"),
+            ),
+        )(self.crashmover_save)
```

Comment:
I wouldn't stomp on the original name. I'd change the new property name to something like `self.crashmover_save_with_retry` or something like that. Ditto for `crashmover_publish` below.

Another way to do this is not make `max_attempts` or `retry_sleep_seconds` configurable things and instead make them constants at the top of the module. Then you could use the retry decorator at the point the two methods are defined. I don't think these are things we're going to want to tune in server configurations.

Reply:
`self.crashmover_save_with_retry` sounds good.

I'd prefer to keep this configurable so the retry tests don't have to block for 10 seconds per test.

Reply:
Oh, I see that now. That makes sense. :+1: